image/manifest.go

```diff
@@ -87,7 +87,22 @@ func (m *manifest) validate(w walker) error {
 	return nil
 }
 
-func (m *manifest) unpack(w walker, dest string) error {
+func (m *manifest) unpack(w walker, dest string) (retErr error) {
+	// error out if the dest directory is not empty
+	if _, err := os.Stat(dest); err == nil {
+		s, _ := ioutil.ReadDir(dest)
```

Comment:
And if you end up using `ReadDir`, you should be checking for errors that it returns and returning those, instead of returning your own “is not empty” error in those cases (e.g. maybe the directory is not empty, but you lack read permission).